### ISSUE TITLE
Add series axis for stacked series chart

### ISSUE THREAD

**Issue by subarroca** [dynatrace-oss/barista#1212]
# Feature Request

## Summary
Charts usually need a value axis and a series axis. Currently `seriesAxis` is incomplete

## Feature Description

Currently stacked series chart shows a valueAxis and series label but we also need a different case where series could be based on a numeric or date value.

### Solution
I suggest adding `seriesAxisMode` with possible values:
- `numberAbsolute`
- `numberPercent`
- `dateRelative`
- `dateAbsolute`

Also I'd introduce `seriesAxisFormat` useful in case of dates but with a default formatting based on DurationFormat.

In the future `visibleLabel` could be merged into `seriesAxisMode` but this would introduce a breaking change

## Attachments

![image](https://user-images.githubusercontent.com/1245057/86013096-0f336380-ba1f-11ea-973a-b59d55d163cb.png)

**Comment by tomheller** [comment]
Hi @subarroca !

I think that `dateRelative` and `dateAbsolute` could benefit from a more clear naming, like `date` and `duration` (if this is what it should mean).

---

This seems to become more and more like a chart that we can already do with highcharts. I'm currently questioning if the stacked-series-chart is exceeding it's boundaries and expectations a bit. Initially it was designed to be a lightweight chart component that can be used in table-cells and such and it has grown quite quickly. 
Maybe we need to reiterate on the scope again. 
With this chart being a derivative of charts that can be achieved with highcharts, we are bound to introduce friction between the behavior of the two.

**Comment by subarroca** [comment]
I agree that this chart is getting more and more complex but as far as I know there's a 'get rid of highcharts' milestone. If the approach is to keep using highcharts then it makes no sense to do this ticket

**Comment by tomheller** [comment]
@subarroca You are right, this has not been communicated as clearly as we wished it would, but the `remove Highcharts` milestone has been postponed indefinitely for now. It does not seem feasible with the current workload and roadmap to work towards a highcharts-less barista-components library. 

If you agree with this, I would suggest to close this ticket, and open another one with a rescoping of the stacked-series-chart to essentially what we see in the _Connected legend_ example of the component. I understand that this would be coming with breaking changes, but I think it is way easier to trim the component back now, as if we will need to do this, once it has a lot of usage. 

We are still trying to build charts that serve a use-case, which is not easily covered by highcharts (sunburst-chart, event-chart, ...).

**Comment by subarroca** [comment]
As we have the need for a chart like this in the series mode, with a legend, that it's something not initially useful for the chart in a table.

We created this chart because we could not make the overlay work correctly, so going back to highcharts is not an option for us. And yes, then I added more options because they made sense with the removal of the lib in mind.

The only thing that I think we could remove is the column mode, but it's just a CSS because the rest of elements are needed and the calculations already done so I suggest closing this ticket but leaving the chart as it is.
